File: S2v2.java
package packageexample.example;

/*
 * Author : Suleiman Al-Othman (Social Media Account => @sulealothman)
 * Ver. 1.0
 * S2 Encryption and Decryption Algorithm method
 */

public class example {
    public static void main(String[] args) {
        
        // Example text
        String msg = "Suleiman Al-Othman";
        
        // Create object from S2v2 Class
        S2v2 Sv2 = new S2v2();
        
        // Using encryption method
        String enc = Sv2.S2Encrypt(msg, "sulealothman");
        
        // Using decryption method
        String dec = Sv2.S2Decrypt(enc, "sulealothman");
        
        
        // Show result
        System.out.println(enc);
        System.out.println(dec);
    }
}

class S2v2 {
    
    // Encryption Method
    public String S2Encrypt(String txt,String password) {
        int lvar = 128 - password.length();
        char passArr[] = password.toCharArray();
        int pCounter = 0;
        int tCounter = 0;
        String result = "";
        for(char i: txt.toCharArray()) {
            if(pCounter == password.length()) pCounter = 0;
            if(tCounter == lvar) tCounter = 0;
            result += (char)(((int)i + (lvar % (tCounter + 1))) + ((int)passArr[pCounter] - pCounter));
            pCounter++;
            tCounter++;
        }
        return result;
    }

    // Decryption Method
    public String S2Decrypt(String txt, String password) {
        int lvar = 128 - password.length();
        char passArr[] = password.toCharArray();
        int pCounter = 0;
        int tCounter = 0;
        String result = "";
        for(char i: txt.toCharArray()) {
            if(pCounter == password.length()) pCounter = 0;
            if(tCounter == lvar) tCounter = 0;
            result += (char)(((int)i - (lvar % (tCounter + 1))) - ((int)passArr[pCounter] - pCounter));
            pCounter++;
            tCounter++;
        }
        return result;
    }

}

File: S2v2.swift
import UIKit
//
//  Author : Suleiman Al-Othman ( Social Media Account => @sulealothman )
//  Ver. 1.0
//  S2 Encryption and Decryption Algorithm method
//

class S2v2 {
    
    // Encryption Method
    public func S2Encrypt(Text:String, Password:String) -> String {
        
        let textArr:[UInt16] = Array(Text.utf16)
        let passArr:[UInt16] = Array(Password.utf16)
        let lVar:Int = 128 - passArr.count
        var tCounter:Int = 0
        var pCounter:Int = 0
        var tempCode:Int? = nil
        var result:String = ""
        
        for chr in textArr {
            if(pCounter == passArr.count) { pCounter = 0 }
            if(tCounter == lVar) { tCounter = 0 }
            tempCode = ((Int(chr) + (lVar % (tCounter + 1))) + (Int(passArr[pCounter]) - pCounter))
            result.append(String(Character(UnicodeScalar(tempCode!)!)))
            tCounter += 1
            pCounter += 1
        }
        return result
    }
    
    // Decryption Method
    public func S2Decrypt(Text:String, Password:String) -> String {
        
        let textArr:[UInt16] = Array(Text.utf16)
        let passArr:[UInt16] = Array(Password.utf16)
        let lVar:Int = 128 - passArr.count
        var tCounter:Int = 0
        var pCounter:Int = 0
        var tempCode:Int? = nil
        var result:String = ""
        
        for chr in textArr {
            if(pCounter == passArr.count) { pCounter = 0 }
            if(tCounter == lVar) { tCounter = 0 }
            tempCode = ((Int(chr) - (lVar % (tCounter + 1))) - (Int(passArr[pCounter]) - pCounter))
            if(tempCode! > -1) { result.append(String(Character(UnicodeScalar(tempCode!)!))) }
            tCounter += 1
            pCounter += 1
        }
        return result
    }
}

// Create object for S2v2 Class
var Sv2 = S2v2()

// Example text
var str = "Suleiman Al-Othman"

// Using encryption method
var enc = Sv2.S2Encrypt(Text: str,Password: "sulealothman")

// Using decryption method
var dec = Sv2.S2Decrypt(Text: enc,Password: "sulealothman")

// Show result
print(enc, dec)